matching-notification-api/src/main/java/org/phenotips/matchingnotification/finder/internal/TestMatchFinder.java

```diff
@@ -0,0 +1,167 @@
+
+import org.phenotips.components.ComponentManagerRegistry;
+import org.phenotips.data.Patient;
+import org.phenotips.matchingnotification.finder.MatchFinder;
+import org.phenotips.matchingnotification.match.PatientMatch;
+import org.phenotips.matchingnotification.match.internal.DefaultPatientMatch;
+import org.phenotips.matchingnotification.match.internal.DefaultPatientMatchUtils;
+import org.phenotips.matchingnotification.match.internal.DefaultPhenotypesMap;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.manager.ComponentLookupException;
+import org.xwiki.component.manager.ComponentManager;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @version $Id$
+ */
+@Component
+@Named("test")
+@Singleton
+public class TestMatchFinder implements MatchFinder
+{
+    private static final String SERVER1 = "SERVER1";
+    private static final String SERVER2 = "SERVER2";
+
+    private static final String Q1 = "Q0000001";
+    private static final String Q1_EMAIL = "mailto:q1@server1.com";
+    private static final String Q1_GENES = "q1_gene1;q1_gene2";
+
+    private static final String Q2 = "Q0000002";
+    private static final String Q2_EMAIL = "mailto:q2@server1.com";
+    private static final String Q2_GENES = "q2_gene1;q2_gene2";
+
+    private static final DefaultPatientMatchUtils UTILS;
+
+    static {
+        DefaultPatientMatchUtils utils = null;
+        try {
+            ComponentManager ccm = ComponentManagerRegistry.getContextComponentManager();
+            utils = ccm.getInstance(DefaultPatientMatchUtils.class);
+        } catch (ComponentLookupException e) {
+        }
+        UTILS = utils;
+    }
+
+    /** */
+    public class TestMatchData
+    {
+        /** */
+        public String referencePatientId;
+        /** */
+        public String referenceServerId;
+        /** */
+        public String matchedPatientId;
+        /** */
+        public String matchedServerId;
+        /** */
+        public Double score;
+        /** */
+        public Double genotypeScore;
+        /** */
+        public Double phenotypeScore;
+        /** */
+        public String href;
+        /** */
+        public String genes;
+        /** */
+        public String matchedGenes;
+        /** */
+        public String phenotypes;
+        /** */
+        public String matchedPhenotypes;
+    }
+
+    private Logger logger = LoggerFactory.getLogger(TestMatchFinder.class);
+
+    @Override
+    public int getPriority() {
+        return 0;
+    }
+
+    @Override
+    public List<PatientMatch> findMatches(Patient patient)
+    {
+        this.logger.debug("Finding test matches for patient {}.", patient.getId());
+
+        List<PatientMatch> matches = new LinkedList<>();
+        if (patient.getId().equals("P0000001")) {
```

Comment:
It's not unit testing, it's used in one of the /src/main classes for now and should be removed eventually.